expression/collation.go

```diff
@@ -152,21 +160,7 @@ var (
 )
 
 func deriveCoercibilityForScarlarFunc(sf *ScalarFunction) Coercibility {
-	if _, ok := sysConstFuncs[sf.FuncName.L]; ok {
-		return CoercibilitySysconst
-	}
-	if sf.RetType.EvalType() != types.ETString {
-		return CoercibilityNumeric
-	}
-
-	_, _, coer, _ := inferCollation(sf.GetArgs()...)
-
-	// it is weird if a ScalarFunction is CoercibilityNumeric but return string type
-	if coer == CoercibilityNumeric {
-		return CoercibilityCoercible
-	}
-
-	return coer
+	panic("this function should never be called")
```

Comment:
this should never be called because the coercibility for `scalarFunction` is set at `newBaseBuiltinFuncWithTp`